src/util/tab-events.js

```diff
@@ -1,5 +1,20 @@
 import eventToPromise from './event-to-promise'
 
+function tabChanged(tabId) {
+    return new Promise((resolve, reject) => {
+        browser.webNavigation.onCommitted.addListener((details)=>{
+            console.log(details);
+            if(details.tabId == tabId && details.frameId == 0)
+                reject();
```

Comment:
My view on this was that, i have to listen to changes in the tabs url coming both from normal redirects like links clicked, and from changes triggered by the HTML 5 History API,  and reject in both cases, but i doubt there is ever a case  where those two can resolve at once and so this can only call reject once
@obsidianart